lib/ueberauth/strategy/google.ex

```diff
@@ -130,15 +125,21 @@ defmodule Ueberauth.Strategy.Google do
     end
   end
 
-  defp with_param(opts, key, conn) do
-    if value = conn.params[to_string(key)], do: Keyword.put(opts, key, value), else: opts
-  end
-
-  defp with_optional(opts, key, conn) do
-    if option(conn, key), do: Keyword.put(opts, key, option(conn, key)), else: opts
+  @doc false
+  defp redirect_options(conn) do
```

Comment:
What I started doing is adding another line to support my specific use case of using "approval=consent" as url parameter. 
IMHO, it makes much more sense to unify all configuration because any of these parameters can be relevant app-wide or per request. The only exception is the state parameter.

Not sure about the implementation quality because I'm new to Elixir but specifically about the to_string, I can only guess it won't add measurable amount of time to the authentication flow.